Source/WarpX.cpp

```diff
@@ -87,6 +87,7 @@ bool WarpX::galerkin_interpolation = true;
 bool WarpX::use_filter        = false;
 bool WarpX::use_kspace_filter       = false;
 bool WarpX::use_filter_compensation = false;
+bool WarpX::use_damp_fields_in_z_guard = false;
```

Comment:
One question based on the recent discussions: should we maybe set this `true` by default for RZ simulations?

Reply:
Probably should default to true, except that the damping should be skipped when periodic_single_box_fft is true (there are no guard cells - how does the ParallelFor handle an empty box?) and when periodic is true (the damped fields are overwritten by the periodic copies so it is a bit of wasted time). Also, if set to true, `PushPSATD` should be fixed so that `DampFieldsInGuards` is only called for RZ, or have it default to true only for RZ.

Reply:
Since `periodic_single_box_fft` requires that the domain is periodic, I'd say it is sufficient to check on periodicity. Personally, I would set the logic here in WarpX.cpp and leave `PushPSATD` as it is. I guess something like this could work?
```cpp
bool WarpX::use_damp_fields_in_z_guard = false;
#ifdef WARPX_DIM_RZ
if (!geom[0].isPeriodic(1))
    WarpX::use_damp_fields_in_z_guard = true;
#endif 
```

Reply:
Yes, this would work, though the `ifdef` block should go just before the parameter is read in, just before line 638 below.

Reply:
The change is committed. Note that I also put the reading of the parameter inside the `ifdef` block since it should only be set when using RZ.